compliance_checker/ioos.py

```diff
@@ -1047,14 +1054,17 @@ def check_wmo_platform_code(self, ds):
 
         valid = True
         ctxt = "wmo_platform_code"
-        msg = "The wmo_platform_code must be a numeric string of 5 or 7 characters"
+        msg = ("The wmo_platform_code must be an alphanumeric string of 5 "
+               "characters or a numeric string of 7 characters")
 
         code = getattr(ds, "wmo_platform_code", None)
         if code:
            if not (
                isinstance(code, str) and
-               code.isnumeric() and
-               (5 <= len(code) <= 7)
+               (
+                   (code.isnumeric() and (len(code)==5 or len(code)==7)) or
+                   re.search(r"^\w{5}$", code)
```

Comment:
`\w` includes underscores, which for the NWS identifier do not form a valid identifier.

Note you can also use `str.isalnum`, but it includes a number of Unicode characters which probably wouldn't be desired.

`isnumeric` and related methods apparently treat Unicode fractions as truthy!  I imagine this is not really what we want, but perhaps we should revisit this later.
https://docs.python.org/3/library/stdtypes.html#str.isnumeric

e.g.  `"ü¾".isalnum()` returns True.
So you might just want to use regex character classes here as so:

`re.search(r"^(?:[a-zA-Z0-9]{5}|[0-9]{7})$", code)`